### PR TITLE
34 ondisconnect should remove user from store

### PR DESCRIPTION
**Pull request by 9j3**

### CODE REVIEW

File: packages/api/src/modules/chat/chat.gateway.ts [hunk 1]
```diff
@@ -38,7 +38,13 @@ export class ChatGateway
   handleDisconnect(client: Socket): void {
     // user is offline
     console.log('DISCONNECT');
-    client.broadcast.emit(`user ${this.users[client.id]} left the channel`);
+
+    // notify all connected client's that there is a new leave
```
Comment (bariod) [inline packages/api/src/modules/chat/chat.gateway.ts:42]:
```suggestion
    // notify all connected clients that there is a new disconnect
```

File: packages/api/src/modules/chat/chat.gateway.ts [hunk 2]
```diff
@@ -64,25 +70,33 @@ export class ChatGateway
     this.users[client.id]['username'] = username;
     this.users[client.id]['isTyping'] = false;
 
+    // notify all connected client's that there is a new connection
```
Comment (bariod) [inline packages/api/src/modules/chat/chat.gateway.ts:73]:
```suggestion
    // notify all connected clients that there is a new connection
```

File: packages/api/src/modules/chat/chat.gateway.ts [hunk 3]
```diff
@@ -64,25 +70,33 @@ export class ChatGateway
...
-      console.log(`Client ${client.id} stopped typing`);
+  @SubscribeMessage('typing:stop')
+  stopTyping(client: Socket) {
+    console.log(`Client ${client.id} stopped typing`);
 
-      client.broadcast.emit('typing:stop', {
-        client: client.id,
-      });
-    }, 1500);
+    // tell other clients that client is typing
```
Comment (bariod) [inline packages/api/src/modules/chat/chat.gateway.ts:96]:
```suggestion
    // tell other clients that client has stopped typing
```

File: packages/app/src/views/HomeView.vue [hunk 4]
```diff
@@ -35,7 +36,16 @@ const sendMessage = () => {
...
-  socket.emit('typing:start');
+  if (isTypingTimeout !== null) {
+    clearTimeout(isTypingTimeout);
+    isTypingTimeout = null;
+  } else {
+    socket.emit('typing:start');
+
+    isTypingTimeout = setTimeout(() => {
+      socket.emit('typing:stop');
+    }, 1000);
```
Comment (bariod) [inline packages/app/src/views/HomeView.vue:47]: Might wanna increase this number eventually. But we'll see in testing.

File: packages/app/src/views/HomeView.vue [hunk 5]
```diff
@@ -95,9 +113,7 @@ socket.on('typing:stop', ({ client }) => {
       <!-- body -->
       <div class="h-full flex">
         <div class="h-full w-64 border-r pt-10 px-5">
-          <p class="text-xs font-medium text-gray-400">
-            TEXT-CHANNELS
-          </p>
+          <p class="text-xs font-medium text-gray-400">TEXT-CHANNELS</p>
```
Comment (bariod) [inline packages/app/src/views/HomeView.vue:116]: Looks like we have a formatting issue once again in the vue files...

**Review by bariod — CHANGES_REQUESTED** [review]

File: packages/app/src/views/HomeView.vue (hunk 5 above)
Comment (9j3) [reply packages/app/src/views/HomeView.vue]: Done

**Review by bariod — APPROVED** [review]

**Pull request merged**